### DISCUSSION TITLE
Get global mesh size

### DISCUSSION THREAD

**Discussion by PG20R** [firedrakeproject/firedrake#2547] [Firedrake support] [answered]
Hi.
Sorry, for this very easy question, but I am a complete beginner and just could not find an answer working for me.
I need the global mesh size for a convergence plot. (In FEniCS one could just type mesh.hmax() for example, but this does not seem to work here). How can I do this for a mesh in Firedrake?

**Comment by joewallwork** [comment, marked as answer]
It depends what you mean by "global mesh size". The following will compute the size of each cell and then give you the maximum. Is that what you are looking for?
```
P0 = FunctionSpace(mesh, "DG", 0)
hmax = project(CellSize(mesh), P0)
print(hmax.vector().gather().max())
```

**Reply by PG20R** [reply]
Thanks, that's what I was looking for.

**Reply by wence-** [reply]
Note this is a non-scalable way of computing the max, you get the entire vector on rank 0. A nicer way:
```python
P0 = FunctionSpace(mesh, "DG", 0)
hmax = project(CellSize(mesh), P0)
with hmax.dat.vec_ro as v:
   global_hmax = v.max()
```
